test/rules/prefer_interpolation_to_compose_strings.dart

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N prefer_interpolation_to_compose_strings`
+
+void main() {
+  String name = 'Juan'+'Pablo'; // OK // It is not the business of this rule
+  int year = 2017;
+  int birth = 1993;
+  'Hello, $name! You are ${year - birth} years old.';
+  'Hello, ' + // LINT
+      name + // LINT
+      '! You are ' + // LINT
+      (year - birth).toString() + // LINT
+      ' years old.';
```

Comment:
Sorry, but I don't agree. One of the biggest UX issues for users of compilers (and linters) is having multiple errors when a single error would suffice. While analyzer fails in many cases, minimizing the number of unnecessary diagnostics is an important goal. I think this really needs to be changed.

(Because of the way the analysis server works, it also has other implications, such as causing duplicated quick fixes to be displayed in the IDE, causing other UX problems.)